File: SDKs/amazonaws.com/ec2/2016-11-15/typescript/src/sdk/models/shared/deletelaunchtemplateresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



// DeleteLaunchTemplateResultLaunchTemplate
/** 
 * Information about the launch template.
**/
export class DeleteLaunchTemplateResultLaunchTemplate extends SpeakeasyBase {
  @SpeakeasyMetadata()
  createTime?: Record<string, any>;

  @SpeakeasyMetadata()
  createdBy?: Record<string, any>;

  @SpeakeasyMetadata()
  defaultVersionNumber?: Record<string, any>;

  @SpeakeasyMetadata()
  latestVersionNumber?: Record<string, any>;

  @SpeakeasyMetadata()
  launchTemplateId?: Record<string, any>;

  @SpeakeasyMetadata()
  launchTemplateName?: Record<string, any>;

  @SpeakeasyMetadata()
  tags?: Record<string, any>;
}


export class DeleteLaunchTemplateResult extends SpeakeasyBase {
  @SpeakeasyMetadata()
  launchTemplate?: DeleteLaunchTemplateResultLaunchTemplate;
}

File: SDKs/amazonaws.com/ec2/2016-11-15/python/src/sdk/models/shared/deletelaunchtemplateresult.py
import dataclasses
from typing import Any,Optional


@dataclasses.dataclass
class DeleteLaunchTemplateResultLaunchTemplate:
    r"""DeleteLaunchTemplateResultLaunchTemplate
    Information about the launch template.
    """
    
    create_time: Optional[dict[str, Any]] = dataclasses.field(default=None)
    created_by: Optional[dict[str, Any]] = dataclasses.field(default=None)
    default_version_number: Optional[dict[str, Any]] = dataclasses.field(default=None)
    latest_version_number: Optional[dict[str, Any]] = dataclasses.field(default=None)
    launch_template_id: Optional[dict[str, Any]] = dataclasses.field(default=None)
    launch_template_name: Optional[dict[str, Any]] = dataclasses.field(default=None)
    tags: Optional[dict[str, Any]] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class DeleteLaunchTemplateResult:
    launch_template: Optional[DeleteLaunchTemplateResultLaunchTemplate] = dataclasses.field(default=None)
    
